ws-server/src/server.rs

```diff
@@ -244,8 +245,15 @@ where
 			tracing::debug!("Accepting new connection: {}", conn_id);
 			let key = {
 				let req = server.receive_request().await?;
-				let host_check = cfg.allowed_hosts.verify("Host", Some(req.headers().host));
-				let origin_check = cfg.allowed_origins.verify("Origin", req.headers().origin);
+
+				let host = std::str::from_utf8(req.headers().host)
+					.map_err(|_e| Error::Custom("Host header must be valid UTF-8".into()))?;
+				let origin = req.headers().origin.and_then(|h| std::str::from_utf8(h).ok());
+
+				let host_check = cfg.access_control.verify_host(host);
+				let origin_check = cfg.access_control.verify_origin(origin, host);
+
+				// TODO: "access-control-request-headers" not possible to fetch via soketto.
```

Comment:
Is this something worthy of making an issue for in soketto?

Reply:
I was wrong, this header only applicable for HTTP Option requests when doing preflight requests.

The websocket  HTTP upgrade request must be `GET` AFAIU.